test/test_autocast.py

```diff
@@ -350,11 +352,30 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def test_mps_autocast_error_message(self):
         with self.assertWarnsRegex(
-            UserWarning, "MPS Autocast only supports dtype of torch.float16 currently."
+            UserWarning,
+            "MPS Autocast only supports dtype of torch.bfloat16 and torch.float16 currently.",
         ):
-            with torch.autocast(device_type="mps", dtype=torch.bfloat16):
+            with torch.autocast(device_type="mps", dtype=torch.float32):
                 _ = torch.ones(10)
 
+    # torch.bfloat16 is only supported on macOS 14 and above.
+    @expectedFailureMPSPre14
+    def test_mps_autocast_bfloat16_supported(self):
+        with torch.amp.autocast(device_type="mps", dtype=torch.bfloat16):
+            x = torch.randn(2, 3, device="mps")
+            y = torch.randn(3, 3, device="mps")
+            result = torch.mm(x, y)
+            self.assertEqual(result.dtype, torch.bfloat16)
+
+    @patch("platform.mac_ver", return_value=("13.0", ("", "", ""), ""))
+    def test_mps_autocast_bfloat16_unsupported(self, mock_mac_ver):
+        with self.assertWarnsRegex(
+            UserWarning,
+            "In MPS autocast, but the target dtype torch.bfloat16 is not supported on macOS versions below 14. Disabling autocast.",
+        ):
+            with torch.amp.autocast(device_type="mps", dtype=torch.bfloat16):
+                _ = torch.ones(10, device="mps")
```

Comment:
This test feels redundant, as previous test will already xfail on older MacsOS and pass on new one

Reply:
Yes, assuming we test on this platform version. I'll remove the test.